public/app/plugins/panel/bargauge/module.tsx

```diff
@@ -61,19 +61,52 @@ export const plugin = new PanelPlugin<Options>(BarGaugePanel)
         defaultValue: defaultOptions.showUnfilled,
         showIf: (options) => options.displayMode !== 'lcd',
       })
-      .addNumberInput({
+      .addRadio({
+        path: 'sizing',
+        name: 'Bar size',
+        settings: {
+          options: [
+            { value: BarGaugeSizing.Auto, label: 'Auto' },
+            { value: BarGaugeSizing.Manual, label: 'Manual' },
+          ],
+        },
+        defaultValue: defaultOptions.sizing,
+      })
+      .addSliderInput({
         path: 'minVizWidth',
         name: 'Min width',
-        description: 'Minimum column width',
+        description: 'Minimum column width (vertical orientation)',
         defaultValue: defaultOptions.minVizWidth,
-        showIf: (options) => options.orientation === VizOrientation.Vertical,
+        settings: {
+          min: 0,
+          max: 300,
+          step: 1,
+        },
+        showIf: (options) => options.sizing === BarGaugeSizing.Manual,
       })
-      .addNumberInput({
+      .addSliderInput({
         path: 'minVizHeight',
         name: 'Min height',
-        description: 'Minimum row height',
+        description: 'Minimum row height (horizontal orientation)',
         defaultValue: defaultOptions.minVizHeight,
-        showIf: (options) => options.orientation === VizOrientation.Horizontal,
+        settings: {
+          min: 0,
+          max: 300,
+          step: 1,
+        },
+        showIf: (options) => options.sizing === BarGaugeSizing.Manual,
+      })
+      .addSliderInput({
+        path: 'maxVizHeight',
+        name: 'Max height',
+        description: 'Maximum row height (horizontal orientation)',
+        defaultValue: defaultOptions.maxVizHeight,
+        settings: {
+          min: 0,
+          max: 300,
+          step: 1,
+        },
+        showIf: (options) => options.sizing === BarGaugeSizing.Manual,
```

Comment:
I think we can still make the manual options a function of layout orientation right?

for auto we show all

for horizontal we show max / min height

for vertical we just show width

With this I think we can keep the description explicitely saying what they are for / apply, which is especially useful in auto mode